network/alsp/manager/manager_helper_test.go

```diff
@@ -0,0 +1,126 @@
+func managerCfgFixture(t *testing.T) *alspmgr.MisbehaviorReportManagerConfig {
+	c, err := config.DefaultConfig()
+	require.NoError(t, err)
+	return &alspmgr.MisbehaviorReportManagerConfig{
+		Logger:                  unittest.Logger(),
+		SpamRecordCacheSize:     c.NetworkConfig.AlspConfig.SpamRecordCacheSize,
+		SpamReportQueueSize:     c.NetworkConfig.AlspConfig.SpamReportQueueSize,
+		HeartBeatInterval:       c.NetworkConfig.AlspConfig.HearBeatInterval,
+		AlspMetrics:             metrics.NewNoopCollector(),
+		HeroCacheMetricsFactory: metrics.NewNoopHeroCacheMetricsFactory(),
+	}
+}
+
+// misbehaviorReportFixture creates a mock misbehavior report for a single origin id.
+// Args:
+// - t: the testing.T instance
+// - originID: the origin id of the misbehavior report
+// Returns:
+// - network.MisbehaviorReport: the misbehavior report
+// Note: the penalty of the misbehavior report is randomly chosen between -1 and -10.
+func misbehaviorReportFixture(t *testing.T, originID flow.Identifier) network.MisbehaviorReport {
+	return misbehaviorReportFixtureWithPenalty(t, originID, math.Min(-1, float64(-1-rand.Intn(10))))
+}
+
+func misbehaviorReportFixtureWithDefaultPenalty(t *testing.T, originID flow.Identifier) network.MisbehaviorReport {
+	return misbehaviorReportFixtureWithPenalty(t, originID, model.DefaultPenaltyValue)
+}
+
+func misbehaviorReportFixtureWithPenalty(t *testing.T, originID flow.Identifier, penalty float64) network.MisbehaviorReport {
+	report := mocknetwork.NewMisbehaviorReport(t)
+	report.On("OriginId").Return(originID)
+	report.On("Reason").Return(alsp.AllMisbehaviorTypes()[rand.Intn(len(alsp.AllMisbehaviorTypes()))])
+	report.On("Penalty").Return(penalty)
+
+	return report
+}
+
+// WithDecayFunc sets the decay function for the MisbehaviorReportManager. Useful for testing purposes to simulate the decay of the penalty without waiting for the actual decay.
+// Args:
+//
+//	f: the decay function.
+//
+// Returns:
+//
+//	a MisbehaviorReportManagerOption that sets the decay function for the MisbehaviorReportManager.
+//
+// Note: this option is useful primarily for testing purposes. The default decay function should be used for production.
+func WithDecayFunc(f alspmgr.SpamRecordDecayFunc) alspmgr.MisbehaviorReportManagerOption {
+	return func(m *alspmgr.MisbehaviorReportManager) {
+		m.DecayFunc = f
+	}
+}
+
+// WithSpamRecordsCacheFactory sets the spam record cache factory for the MisbehaviorReportManager.
+// Args:
+//
+//	f: the spam record cache factory.
+//
+// Returns:
+//
+//	a MisbehaviorReportManagerOption that sets the spam record cache for the MisbehaviorReportManager.
+//
+// Note: this option is useful primarily for testing purposes. The default factory should be sufficient for production.
+func WithSpamRecordsCacheFactory(f alspmgr.SpamRecordCacheFactory) alspmgr.MisbehaviorReportManagerOption {
+	return func(m *alspmgr.MisbehaviorReportManager) {
+		m.CacheFactory = f
+	}
+}
```

Comment:
```suggestion
```
see https://github.com/onflow/flow-go/pull/4485/files#r1259083046

Reply:
I suggest keeping them as it is, they must be solely used for testing.

Reply:
Moved these back to ALSP Manager (`network/alsp/manager/manager.go`)
f7b7b62c4c696eab9f87a66fd25a77df71f8cdad

2e027bc40f1d064bf0a28ebc783153c5b97ea240